apps/clock/js/alarm.js

```diff
@@ -156,6 +156,81 @@ define(function(require, exports, module) {
       // Only update the application if this alarm was actually saved
       // (i.e. it has an ID).
       if (this.id) {
+        var self = this,
+            alarmDatabase = require('alarm_database');
+        // Updating Alarm info in the DataStore
+        var storeAlarms = [];
+        alarmDatabase.getAll().then((alarms) => {
+          // Logging in a loop to ensure we don't overrun the line buffer:
+          var now = new Date();
+          function addZero(i) {
+            if(typeof i == "String") {
+              i = parseInt(i);
+            }
+            if (i < 10) {
+              i = "0" + i;
+            }
+            return i + "";
+          }
+          alarms.forEach(function(a) {
+            // Grabbing alarms that are 'normal'
+            if(a.registeredAlarms.normal) {
+              //Check for Repeated Alarms
+              if(Object.keys(a.repeat).length != 0) {
+                var hasToBePushed = false;
+                for (var n in a.repeat) {
+                  var day = n;
+                  if(now.getDay() == 6 && n < 6) {
+                    day = 7 + parseInt(n);
+                  }
+                  if(day == now.getDay() || day == now.getDay() + 1) {
```

Comment:
I don't think `getNextAlarmFireTime()` does what it is supposed to do. If I have 4 alarms set in the clock app, say 1st one for 2 days from now, and others for 10 min from now, it always returns the first alarm, even though it is supposed to set off 2 days from today.
Or may be am I using it wrong?